R/method.R

```diff
@@ -36,11 +36,12 @@
 #'
 #' # But it can be useful to explicitly retrieve a method in order to
 #' # inspect its implementation
-#' method(bizarro, list("numeric"))
-#' method(bizarro, list("factor"))
+#' method(bizarro, "numeric")
+#' method(bizarro, s3_class("factor"))
 method <- function(generic, signature) {
-  signature <- as_signature(signature)
+  # TODO: check that signature doesn't contain any unions
 
+  signature <- as_signature(signature)
```

Comment:
It may be worth checking that at least 1 dispatch arg is supplied in the signature, because currently you can do:

``` r
library(R7)

bizarro <- new_generic("bizarro", dispatch_args = "x")

method(bizarro, list())
#> Error in signature[[1]]: subscript out of bounds
```

<sup>Created on 2022-02-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
Or maybe it is a bug in `method_impl()` because that is where `signature[[1]]` is called

Reply:
Now at https://github.com/RConsortium/OOP-WG/issues/146